c++/src/kj/mutex.c++

```diff
@@ -850,6 +850,15 @@ void Once::reset() {
   InitOnceInitialize(&coercedInitOnce);
 }
 
+#elif defined(KJ_NO_THREADS)
+
+Mutex::Mutex() {}
+Mutex::~Mutex() {}
+bool Mutex::lock(Exclusivity exclusivity, Maybe<Duration> timeout, NoopSourceLocation) {
+  return true;
```

Comment:
Perhaps in no-threads mode, `kj::Mutex` should still use a boolean to track whether it is locked, and abort if a recursive lock is attempted? (Such a recursive lock would deadlock when threads are enabled.)

Reply:
I like that!